### PR TITLE
Music: spinner

### PR DESCRIPTION
**Pull request by breville**
This adds a spinner to the `LabContainer` which wraps Music Lab.  The spinner fades in after a 2 second delay, so that it doesn't appear for users with particularly fast connections.  Some helper text fades in after a further 8 seconds.

Given that Music Lab can change levels without a page reload, this all works for both first load and subsequent level switches.

https://github.com/code-dot-org/code-dot-org/assets/2205926/288554cc-4312-4345-b57f-e0980ed80700

### CODE REVIEW

File: apps/src/code-studio/components/LabContainer.module.scss [hunk 1]
```diff
@@ -33,7 +66,24 @@
...
+
+    .slowLoadContainer {
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+      gap: 20px;
+
+      .spinnerContainer {
+        color: $neutral_dark40;
+        animation: fade-in-spinner-container 3s;
```
Comment (molly-moen) [inline apps/src/code-studio/components/LabContainer.module.scss:80]: I didn't know you could do this in css!

**Review by molly-moen — APPROVED** [review]
LGTM! This taught me some new css tricks :)

**Review by molly-moen — APPROVED** [review]

File: apps/i18n/common/en_us.json [hunk 2]
```diff
@@ -1210,6 +1210,7 @@
   "keyValuePairsHeader": "KEY/VALUE PAIRS",
   "keyValuePairLink": "Key/value pairs",
   "kindergarten": "Kindergarten",
+  "labContainerSlowLoading": "This is taking longer than usual...",
```
Comment (sanchitmalhotra126) [inline apps/i18n/common/en_us.json]: Should this just be `slowLoading`? That would match the server-side string key and could potentially be reused

**Review by sanchitmalhotra126 — APPROVED** [review]
Very cool! I think personally I find the 2 seconds to be a bit slow, where it seems like the page instead had a delay loading the spinner. Maybe a shorter value would work better? But either way, code looks good!

File: apps/i18n/common/en_us.json (hunk 2 above)
Comment (breville) [reply apps/i18n/common/en_us.json]: Fair call.  Will update.

**Comment by breville** [comment]
> I think personally I find the 2 seconds to be a bit slow, where it seems like the page instead had a delay loading the spinner. Maybe a shorter value would work better?

You're probably right, and definitely happy to make adjustments as we experience this on a variety of environments.  I might leave as-is for now, just because I have such an aversion to spinners that I'm curious how much we can get away without showing one :).

**Pull request merged**